frontend/src/pages/boardsPage/BoardPage.jsx

```diff
@@ -0,0 +1,205 @@
+import Board from "../../components/boardComponent/Board";
+import NewBoard from "../../components/addBoardModal/NewBoardModal";
+import AuthModal from "../../components/authModal/AuthModal";
+
+// router functions
+import { Navigate, useLocation } from "react-router-dom";
+
+
+// could get this to outside helper function
+async function getAllBoardsFiltered(term, id){
+  const DATABASE_URL = import.meta.env.VITE_DATABASE_URL
+
+  let url = `${DATABASE_URL}/filter/${term}`
+  if (term == "my boards"){
+    url = `${DATABASE_URL}/filter/myboards/${id}`
+  }
+
+  const options = {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+    }
+  }
+
+  const allBoards = await fetch(url, options)
+  const resJson = await allBoards.json()
+
+  return resJson
+}
+
+async function getSearchedBoards(term){
+  const DATABASE_URL = import.meta.env.VITE_DATABASE_URL
+
+  const url = `${DATABASE_URL}/board/search/${term}`
+
+  const options = {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+    }
+  }
+
+  const allBoards = await fetch(url, options)
+  const resJson = await allBoards.json()
+
+  return resJson
+}
+
+async function addNewBoardApi(board){
+  const DATABASE_URL = import.meta.env.VITE_DATABASE_URL
+
+  const url = `${DATABASE_URL}/board/`
+
+
+  const options = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(board)
+  }
+
+  const allBoards = await fetch(url, options)
+  const resJson = await allBoards.json()
+  return resJson
+}
+
+function BoardPage (props) {
+  // stores search term and filter option
+  const location = useLocation(); // initialize location for Navitate()
+
+
+  const [searchTerm, setSearchTerm] = useState(location.state? location.state.searchTerm : "")
+  const [filter, setFilter] = useState(location.state? location.state.filterTerm : "all")
+  const [allBoards, setAllBoards] = useState([])
+  const [showNewBoard, setShowNewBoard] = useState(false)
+  const [cardPage, setCardPage] = useState(null)
+  const [showLogin, setShowLogin]  = useState(false)
+  const [deleted ,setDeleted] = useState(0)
+
+  // user states
+  const user = props.user
+  const setUser = props.setUser
+
+  // stores board visuals for bottom
+  useEffect(() => {
+
+    if (filter) {
+      setSearchTerm("")
+
+      if (!user.id && filter=="my boards"){
+        alert("please login first")
+        return;
+      } else if (user.id && filter=="my boards") {
+        getAllBoardsFiltered(filter.toLowerCase(), user.id)
+        .then(data => {
+          if (!data.error) {
+            setAllBoards([])
+            setAllBoards(data)
+          }
+        })
+
+        return;
+      }
+
+
+      getAllBoardsFiltered(filter.toLowerCase())
+        .then(data => {
+          if (!data.error) {
+            setAllBoards(data)
+          }
+        })
+    }
+
+  }, [filter, deleted])
+
+  const setCardPageDetails = (boardDetails) => {
```

Comment:
I believe we don't need this function, we can direction put `setCardPage` into `<Board key={board.id} setCardPage={setCardPage} img={board.mediaUrl} description={board.description} board_id={board.id} cards={board.cards} authorId={board.authorId} setDeleted={setDeleted} user={user} title={board.title}/>`

Similar idea for the other hooks!